addon/index.js

```diff
@@ -186,6 +188,8 @@ function installExperiment(experiment) {
 }
 
 function uninstallSelf() {
+  // HACK: onUnload often doesn't seem to get called, here.
+  exports.onUnload('uninstall');
```

Comment:
I don't know what the cases are - I can't reproduce it reliably. About half the time, that code never runs for me on dev edition. That was what caused this issue originally for me, because the `addon-self:uninstalled` didn't always get sent and the `store` data didn't get deleted.